xarray/core/dataarray.py

```diff
@@ -1155,6 +1156,8 @@ def chunk(
         token: str | None = None,
         lock: bool = False,
         inline_array: bool = False,
+        manager: Literal["dask", "cubed"] = "dask",
```

Comment:
How about using `like` like in numpy?

Reply:
I think using a preposition (e.g. "like", "via") would be nice for the user-facing code, but if we have to use it internally then passing around a `like` kwarg would be confusing, so I probably still need to think of a name to use inside xarray.

Reply:
It may be useful to think of how to read in an xarray object using cubed. we use `parallel=True` for dask now, so perhaps the kwarg could be `parallel`?

Alternatively, we could follow our current practice of registering accessors so we could have `.cubed.chunk`? Are there other methods that would be useful?

Reply:
> It may be useful to think of how to read in an xarray object using cubed. we use parallel=True for dask now, so perhaps the kwarg could be parallel?

`parallel` in `open_mfdataset` does something a bit different though right? It's about whether or not you do the opening in parallel with `dask.delayed`, not whether the returned data is as a parallel-capable array type.

> Alternatively, we could follow our current practice of registering accessors so we could have .cubed.chunk?

We could do this, maybe we should. But I feel like that's mainly an advantage if it allows us to completely avoid changes to xarray, so cubed is never special-cased in our codebase. However...

> Are there other methods that would be useful?

Generally we need to enabled cubed as an option in `open_dataset`, `.chunk`, `.compute/load`, `apply_ufunc`, and maybe `map_blocks`. That should be it, but I don't think all of these changes can be covered by duck-typing + an accessor - particularly inside `apply_ufunc` (because you can't fetch `dask.array.apply_ufunc` from `dask.array.Array`). It's more analogous to defining a backend entrypoint imo.

On the other hand a cubed accessor could work out neatly - both the accessor and the tests of cubed-xarray integration could live in cubed's repo (or a new `cubed-xarray` repo).

Reply:
> It's about whether or not you do the opening in parallel with dask.delayed, not whether the returned data is as a parallel-capable array type.

We can't concatenate lazy arrays so it ends up creating dask arrays.

> On the other hand a cubed accessor could work out neatly - both the accessor and the tests of cubed-xarray integration could live in cubed's repo (or a new cubed-xarray repo).

Yeah this is the nice thing. cupy-xarray is pretty tiny, but will be a lot more valuable once we set up duck array tests for it.

>  kind of want to think about this as a new form of "plugin" point,

:+1: We also directly call `dask.array.map_blocks` in a number of places (see `.dt` accessor for example), so we'd need that too.

Reply:
Just an idea: what about using None as default which uses dask if installed and falls back to cubed?
Then in the case that only cubed is installed you don't need to set it?

Reply:
> Then in the case that only cubed is installed you don't need to set it?

Dask is a dependency of cubed so I don't think this situation can occur.

Reply:
Nvmd then :)

Reply:
Sorry I missed the discussion today. Did you pick a way to proceed? 

I realized that it might be nice to create `duck_parallel_ops.py` for `apply_gufunc` and `map_blocks` so that all the special-casing can be somewhat self-contained.

Reply:
> I realized that it might be nice to create duck_parallel_ops.py for apply_gufunc and map_blocks so that all the special-casing can be somewhat self-contained.

That's actually pretty much what we decided today :slightly_smiling_face: A separate module to contain this stuff internally for now, so we can generalise the calls in the main codebase without yet having to think about what other libraries should do to register their parallel backends.

The point was also made that it's just a namespace problem, so if `dask.Array` linked to the `apply_gufunc` function via `dask.Array.apply_gufunc` (and so did cubed), then we wouldn't need to register anything else beyond requiring that an array type which defined `.chunks` also gave access to `.apply_gufunc`. But we can punt on this for now.

Reply:
Sounds great!

Reply:
@dcherian I [had a go](https://github.com/pydata/xarray/pull/7019/commits/c616a85dacd374fe6b9063041dcec1fd8e18a67d) at making an adapter class to factor out the special casing into a new file `parallelcompat.py`. It's only meant to be a temporary implementation (and certainly not public yet) but wondering if it looked okay to you as a general approach?